### DISCUSSION TITLE
moodle

### DISCUSSION THREAD

**Discussion by nsrmhjl** [nsrmhjl/ptDtSc#2] [Q&A]
<div dir='rtl'>
سلام آقای @AlirezaDoustmohammadi خوبی؟
میشه بهم یاد بدی برای گرفتن هاست و راه انداختن moodle چیکار باید بکنم؟
باید پیش از این کلاس اون رو راه بندازم تا دانشجوهه اول اونها رو دوباره مرور کنه.

**Comment by AlirezaDostmohammadi** [comment]
سلام آقای دکتر. وقتتون بخیر. امیدوارم حالتون خوب باشه. بله حتما در خدمتتون هستم. 
آقای دکتر حجم فایل فایل های بک آپ مودلی که دارید حدودا چقدره؟ مثلا یک هاست با 5 تا 10 گیگ حافظه کافی هست؟
و سرعت هاست مهمه براتون؟ میشه هاست با سرعت هارد معمولی (هارد های HDD گرفت که ارزون تر در میاد ولی سرعت بالا اومدن سایت قطعا پایین میاد)
یا هاست پرسرعت تر گرفت (هاست با هارد NVME).

اگر هاست با سرعت معمولی بخوایم بگیریم:
پلن 6 و 7 سایت میهن وب هاست مناسب هست:

https://my.mihanwebhost.com/cart.php?gid=1

اگر هاست با سرعت بالا بخوایم بگیریم:
پلن مقدماتی سایت parsvds مناسب هست.

https://parsvds.com/host/wordpress/

البته نیاز به یک دامنه هم داریم که دامنه های .ir سالانه 16 هزار تومن هزینه شه.

**Reply by AlirezaDostmohammadi** [reply]
گرفتن هاست و دامنه هم حدود ۳ روز کاری طول میکشه. چون دامنه دات آی را بعد از خرید نیاز به امضا یک سری تعهدنامه ها داره که موجب میشه روند کار طولانی بشه. البته میشه دامنه های دیگه گرفت مثل دات کام ولی هزینه اش خیلی بیشتره.

**Comment by AlirezaDostmohammadi** [comment]
آقای دکتر البته من برای خودم هم یک هاست و دامنه دارم که الان نگاه کردم 300 مگابایت حدودا الان 2 گیگ اش خالی هست. اگر همین مقدار مناسب هست از اینم میتونیم استفاده کنیم

**Comment by nsrmhjl** [comment]
<div dir='rtl'>
سلام 
احوال شما؟
ممنون از شما بابت اطلاعات.

قبلا روی این سایت گذاشته بودمش:
</div>

[gnomio](https://www.gnomio.com/)

<div dir='rtl'>
فکر کنم چون ویدیو داره، بهتر هست که سرعتش بالاتر باشه. نظر شما چیه؟

حجم فایل ۵۰۰ مگ هست. ولی می خوام خودم این کارارو بکنم و یاد بگیرم. 

خود اون هاست برام دومین هم میگیره؟ 

کلا من تو این چیزا خیلی یول هستم.
</div>

**Reply by AlirezaDostmohammadi** [reply]
سلام دکتر جان. سلامت باشید. خیلی ممنونم. شما خوبید؟ اختیار دارید. 
بله آقای دکتر کلا استفاده از هاست های نسل قدیمی که از هارد های HDD استفاده میکنن کاربرد چندانی نداره و هاست با هارد NVMe خیلی بهتر هست. 
با توجه به میزان حجمی که فرمودید پلن مقدماتی parsvds مناسب هست و هزینه خیلی مقرون به صرفه ای در میاد. قیمت اش 354 هزار تومان سالیانه در میاد و سرعت هارد اش هم بالا است. 
شرکت های دیگه هم هستن که هاست با هارد nvme ارایه میدن اما بین معروف هاشون مثل ایران سرور و میهن وب هاست و پارس وی دی اس من از پارس وی دی اس هم از لحاظ امکاناتی که میده و هم قیمت بیشتر راضی بودم.
باتوجه به اینکه در حال حاضر هیچ تفاوت قیمتی وجود نداره که هاست از ایران یا خارج ایران بگیریم (شرکت های اراده دهنده سرورها کلا با قیمت دلار خدمات ارایه میدن و به همین دلیل قیمت هاست ایرانی و خارجی در پلن های اولیه که میزان حجم هاست بالا نیست یکی شده) بهتره هاست از فرانسه بگیریم که سرعت بالاتری داره و  همون 354 هزار تومن در میاد.

**Reply by AlirezaDostmohammadi** [reply]
برای تهیه هم کافیه ابتدا در سایتش ثبت نام کنید
https://my.parsvds.com/register.php
بعد به پشتیبانی تیکت بزنید که هاست پلن مقدماتی از فرانسه می خواید و بفرمایید که فاکتور بدون احتساب 9 درصد مالیات بزنن و شماره کارت بدن تا پرداخت بشه.
علت گرفتن شماره کارت همین بحث مالیات هست. پرداختی میشه به صورت آنلاین از طریق سایت شون انجام داد که فاکتور رسمی داده میشه و 9 درصد مالیات میخوره و یا از طریق شماره کارت شرکت شون که اون 9 درصد مالیات حذف میشه و فاکتور غیر رسمی داده میشه.
پلن هاشون هم به این شکل هست:
https://parsvds.com/host/cpanel/

**Comment by AlirezaDostmohammadi** [comment]
در رابطه با دامنه آقای دکتر ثبت دامنه توسط شرکت ها صورت میگیره.
ابتدا باید داخل این بخش سایت اسم مدنظرتون بزنید تا ببینید چه دامنه هایی براش آزاد هست.
https://parsvds.com/domain/
بعد میتونید انتخاب کنید که مثلا این اسم با دامنه دات آی آر براتون تهیه بشه.
اگر صرفا برای کار کلاسی از هاست تون قصد دارید استفاده کنید دامنه دات آی آر کافی هست و قیمت خیلی پایینی داره (سالانه 16 هزار تومن و میشه یک ساله یا چهارساله خریداری اش کرد)
ولی برای کار تجاری دامنه دات کام خیلی بهتره (هزینه اش: سالیانه 425 هزار تومن ) هم از این لحاظ که احیانا اگر روزی تحریمی برای دامنه های دات آی آر در نظر گرفته بشه -بیزنسی که راه افتاده لطمه نمیبینه و هم اینکه حدود 8 ساعت بعد خرید دامنه در اختیار تون قرار میگیره
اما دامنه دات آی آر به این شکل نیست. ابتدا باید در سایت ایرنیک به نشانی 
https://www.nic.ir/
ثبت نام کنید و  آی دی ای که این سایت در اختیار شما قرار میده رو برای شرکت ارایه دهنده هاست یعنی parsvds بفرستید و بعد اونها درخواست دامنه میدن.
بعدش برای شما چند تا نامه تعهدنامه از طرف ایرنیک در رابطه با دامنه نهیه شده ارسال میشه که باید امضا و آپلود اش کنید و بعد دامنه تون فعال میشه.
این فرایند در بهترین حالت ۳ روز کاری زمان میبره.

**Reply by nsrmhjl** [reply]
<div dir='rtl'>
سلام آقای @AlirezaDoustmohammadi 
احوال
خیلی ممنون.
به نطرت .net چطوره؟

**Reply by nsrmhjl** [reply]
<div dir='rtl'>
برای دومین نمیشه مالیات ندیم؟

**Reply by nsrmhjl** [reply]
<div dir='rtl'>
این دامنه رو خریدم:
</div>
mohajelresearch.net
<div dir='rtl'>
مالیات هم میگن که نمیشه نگیریم. مثل اینکه مچشون رو گرفتن.

**Reply by AlirezaDostmohammadi** [reply]
اختیار دارید دکتر جان. دامنه تازه خریداری کردید درسته؟

**Reply by AlirezaDostmohammadi** [reply]
چون فعال نیست. حدود ۷ تا ۸ ساعت طول میکشه فعال بشه و توسط DNS ها شناسایی بشه

**Comment by AlirezaDostmohammadi** [comment]
سلام آقای دکتر. وقتتون بخیر. سلامت باشید. خیلی ممنونم. امیدوارم حالتون خوب باشه.
فرقی نداره .net هم خوبه. فقط .ir عه که اذیت میکنه

**Comment by AlirezaDostmohammadi** [comment]
چقدر عجیب. من یکی از دوستانم حدود 10 روز پیش به همین شکل خرید کرد ازشون و مشکلی نبود. شماره کارت میدادن و پول به حساب شون واریز میشد

**Reply by nsrmhjl** [reply]
شاید از قیافم خوششون نیومده!!

**Reply by AlirezaDostmohammadi** [reply]
اختیار دارید دکتر جان. 
دامنه تازه گرفتید درسته؟
جون فعال نیست و حدود ۷ تا ۸ ساعت طول میکشه که دامنه روی DNS ها ست بشه

**Reply by nsrmhjl** [reply]
<div dir='rtl'>
دیروز ساعت ۷ عصر خریدمش اول می خوستم بذارم mohajellab.net بعد دیدم ll خیلی جالب نیست. بعد گذاشتم mohajelresearch.net ولی الان فکر می کنم که کاش می ذاشتم mohajelslab.net.

**Reply by nsrmhjl** [reply]
هاست رو هم گقتن که فردا جواب میدن همون پلان مقدماتی فرانسه

**Comment by AlirezaDostmohammadi** [comment]
آقای دکتر هر وقت هاست به پنل تون اضافه شد من در خدمتتون هستم درباره بحث
مودل.
فقط آنلاین درباره محیط هاست و مودل صحبت کنیم یا حضورا خدمت برسم؟

**Reply by nsrmhjl** [reply]
هر جور شما راحتی
حتی اگه فکر می کنی خودم از پسش بر میام، برام اینجا بنویس خودم انجامش بدم.

**Comment by nsrmhjl** [comment]
<div dir='rtl'>
سلام آقای @AlirezaDoustmohammadi 
خوبی؟
پنل رو دادن دستم. 
حالا باید چیکار کنم؟
</div>

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر جان خوبید؟
وقتتون بخیر

پوزش از بابت تاخیر من

**Comment by AlirezaDostmohammadi** [comment]
https://drive.google.com/file/d/1VSJaxPf_aqH1AdcL05JNlKN15-fX7LTi/view?usp=sharing

**Comment by AlirezaDostmohammadi** [comment]
خدمت شما آقای دکتر. جساراتا داخل ویدیو به طور کامل خدمتتون عرض کردم. جسارت من به بزرگی خودتون ببخشید بابت ضبط ویدیو. حس کردم اینطور بهتر میتونیم کار ببریم جلو

**Reply by nsrmhjl** [reply]
این چه حرفیه؟ خیلی لطف کردی. خیلی بهتر هست. فردا اجراش می کنم و خبر میدم.

**Comment by AlirezaDostmohammadi** [comment]
سلامت باشید دکتر جان

**Comment by nsrmhjl** [comment]
![image](https://user-images.githubusercontent.com/58214153/184630370-1e0cb052-04e1-405b-b399-6144c53bfa1f.png)

**Reply by nsrmhjl** [reply]
موقع انکریپت کردن این ارور رو میده

**Comment by AlirezaDostmohammadi** [comment]
سلام آقای دکتر وقتتون بخیر. فکر میکنم علتش اینکه دامنه توسط شرکت به درستی
روی DNS قرار نگرفته. فکر میکنم باید بهشون تیکت بزنید تا این مورد رو اصلاح
کنند

**Reply by nsrmhjl** [reply]
فرستادم. 
چطوری می تونم مودل رو تو همچین آدرسی نصب کنم:
edu.mohjelresearch.net

**Comment by AlirezaDostmohammadi** [comment]
<div dir='rtl'>
آقای دکتر آدرسی که ارسال فرمودید sub domain هست. بنابر این ابتدا باید sub
domain بسازید. در صفحه اول محیط Cpanel دکمه ای وجود داره که نوشته subdomains
</div>

[image: Screenshot from 2022-08-15 08-47-02.png]

<div dir='rtl'>
سپس در صفحه جدید که باز میشه کافیه تنها subdomainیعنی همون edu رو بنویسید.
</div>

[image: Screenshot from 2022-08-15 08-47-27.png]

<div dir='rtl'>
بعد وارد file manager باید بشید و تمام فایل های نصب مودل رو به فولدر edu که
در public_htmlقرار داره منتقل کنید.
</div>

[image: Screenshot from 2022-08-15 08-48-28.png]

**Comment by AlirezaDostmohammadi** [comment]
آقای دکتر چون تصاویر تو گیتهاب نمیاد، به ایمیل تون این پیامم رو فرستادم

**Reply by nsrmhjl** [reply]
<div dir='rtl'>
سایت راه افتاد. حالا مونده آپلود بک آپ سایت قبلی
<div dir='rtl'>

![image](https://user-images.githubusercontent.com/58214153/184645110-e27221aa-c18a-4e14-842f-cb2c7d8b0da8.png)

**Reply by nsrmhjl** [reply]
چرا عکس نمیاد؟

**Comment by AlirezaDostmohammadi** [comment]
بسیار عالی. کلا تو گیتهاب انگار عکس نمیشه فرستاد.

**Reply by nsrmhjl** [reply]
<div dir='rtl'>
سلام @AlirezaDoustmohammadi 
خوبی؟
این ارور رو موقع آپلود میده:
</div>

![image](https://user-images.githubusercontent.com/58214153/184863452-e357cf0f-c93b-4ac5-9298-1867eaa63516.png)

**Reply by nsrmhjl** [reply]
من تو فرستادن عکس مشکلی ندارم. فکر کنم مربوط به اسکرین شات تو اوبونتو میشه.

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر جان. خیلی ممنون. شما خوبید؟
حجمش فایل چقدر هست؟

**Reply by nsrmhjl** [reply]
<div dir='rtl'>
495 مگ 
تو خود سرور آپلودش کردم. یعنی تو publicurl اما نمی دونم کجا بذارمش که بتونم به خورد moodle بدمش. 
میتونی امروز بیای؟
</div>

**Reply by nsrmhjl** [reply]
<div dir='rtl'>
چطور می تونم حد سایز فایل رو تو php افزایش بدم؟
</div>

![image](https://user-images.githubusercontent.com/58214153/185055685-7271bd82-4ad6-4259-a50a-9acc9854b030.png)

**Comment by AlirezaDostmohammadi** [comment]
سلام آقای دکتر خوبید؟

لطفا داخل cpanel تو بخش سرچ، سرچ بفرمایید:
MultiPHP INI Editor

در صفحه ای که باز میشه ساب دایرکتورب تون لطفا انتخاب کنید.

[image: Screenshot from 2022-08-17 05-20-46.png]

دو گزینه هست باید تغییر بدید.
یکی
max_execution_time هست که پیش فرض 300 ثانیه است و بهتره تغییرش بدید و تایم
خیلی بیشتر بشه چون اینترنت ایران خیلی کنده.
یکی upload_max_filesize هست که پیش فرض 128 مگابایت هست و باید حداقل به
میزان اندازه فایل backup تون زیاد بشه.

بعدش لطفا مجدد فایل backup  از طریق مودل آپلود کنید.

**Reply by nsrmhjl** [reply]
<div dir='rtl'>
تغییرش دادم اما تو مودل تغییر نکرده. به نظرت باید موودل دوباره نصب بشه؟

**Reply by nsrmhjl** [reply]
اگر <'div dir='rtl> بذاری راست چین میشه و راحت تر میشه خوند پیغام ها رو

**Reply by nsrmhjl** [reply]
<div dir='rtl'>
درست شد لیمیت
بذار ببینم آپلود میشه

**Reply by AlirezaDostmohammadi** [reply]
<'div dir='rtl>
خیلی ممنون آقای دکتر. فکر میکنم نیاز به نصب مجدد مودل نیست و باید تغییرات بشینه تو مودل. در رابطه با کلا تمام گزین هایی که در بخش ساب دامنه MultiPHP INI Editor وجود داشت لطفا دو بخش max_execution_time و upload_max_filesize تغییر بدید. اگر تغییرات ننشست احتمالا نیاز به نصب مجدد هست. 
چون هاست رو نمیشه ریستارت کرد تا تغییرات مطمن بشیم اعمال شده

**Reply by nsrmhjl** [reply]
<div dir='rtl'>
بله برای همه ساب دومین ها تغییر دادم و اعمال شد. الان در حال آپلود هست.
</div>

[اینجا](https://moodle.org/mod/forum/discuss.php?d=328139
)
<div dir='rtl'>
گفته خیلی امیدی نیست.
</div>

**Comment by AlirezaDostmohammadi** [comment]
<'div dir='rtl>

آقای دکتر من چیزی که فهمیدم اینه کلا فایل رو به مودل نمیشه فهموند از طریق سرور و امکان خطا هم داره و باید حتما از طریق خود مودل آپلود بشه.
این دفعه اگر آپلود بشه، برای آینده اگر سرور مثل الان مال خودتون باشه مشکلی نخواهید داشت چون به جای بک آپ گرفتن از مودل، از هاست بک آپ میگیریم و خیلی راحت تر میشه اطلاعات restore کرد

**Reply by nsrmhjl** [reply]
درست شد.
خیلی ممنون.
تو مودل راهی هست که بشه یه فایل ژوپیتر رو امبد کرد یا یه پایتون اسکریپت رو و بعد به پایتون توی کامپیوتر خودمون وصلش کرد؟

**Comment by AlirezaDostmohammadi** [comment]
سلامت باشید دکتر جان.
یه توضیحی بدین شکل پیدا کردم:

https://github.com/jupyterhub/ltiauthenticator/blob/main/README.md#moodle

**Reply by nsrmhjl** [reply]
<div dir='rtl'>
نظرت راجع به این چیه؟
</div>

[Developing_Programming_Courses_with_Moodle_and_VPL-evenpaged-preface-nodate-excerpts.pdf](https://github.com/nsrmhjl/ptDtSc/files/9376874/Developing_Programming_Courses_with_Moodle_and_VPL-evenpaged-preface-nodate-excerpts.pdf)

**Comment by AlirezaDostmohammadi** [comment]
<div dir='rtl'>
سلام آقای دکتر وقتتون بخیر. خوبید؟ روی لینک میزنم چیزی برای من باز نمیشه متاستفانه

</div>

**Reply by nsrmhjl** [reply]
https://www.researchgate.net/publication/288944522_Developing_Programming_Courses_with_Moodle_and_VPL_-_The_Teacher's_Guide_to_the_Virtual_Programming_Lab

**Comment by AlirezaDostmohammadi** [comment]
<div dir='rtl'>
آقای دکتر من با این پلاگین کار نکردم اما تو صفحه مودل دیدم به نظر پلاگین جالبیه.
تو این صفحه  پلاگین های خوب مودل برای course های برنامه نویسی هست:
https://moodle.org/plugins/?q=programming

</div>
